Plugins/Geant4/include/Acts/Plugins/Geant4/Geant4Converters.hpp

```diff
@@ -0,0 +1,152 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Common.hpp"
+
+#include <memory>
+#include <tuple>
+
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+#include "G4Transform3D.hh"
+
+class G4Box;
+class G4Material;
+class G4Trd;
+class G4Trap;
+class G4Tubs;
+class G4VSolid;
+class G4VPhysicalVolume;
+
+namespace Acts {
+
+struct Geant4AlgebraConverter {
+  // A potential scalar between Geant4 and ACTS
+  ActsScalar scale = 1.;
+
+  /// @brief  Translate a geometry transform: translation only
+  ///
+  /// @param g4Trans the translation of the Geant4 object
+  ///
+  /// @return a Acts transform
+  Transform3 transform(const G4ThreeVector& g4Trans);
+
+  /// @brief  Translate a geometry transform
+  ///
+  /// @param g4Rot the rotation of the Geant4 object
+  /// @param g4Trans the translation of the Geant4 object
+  ///
+  /// @return a Acts transform
+  Transform3 transform(const G4RotationMatrix& g4Rot,
+                       const G4ThreeVector& g4Trans);
+
+  /// @brief  Translate a geometry transform
+  ///
+  /// @param g4Trf theGeant4 transform object
+  ///
+  /// @return a Acts transform
+  Transform3 transform(const G4Transform3D& g4Trf);
+};
+
+class AnnulusBounds;
+class CylinderBounds;
+class RadialBounds;
+class RectangleBounds;
+class TrapezoidBounds;
+class PlanarBounds;
+
+// The following set of converters convert a Geant4 volume shape
+// to an ACTS surface bounds object, this is for converting the volume
+// based geometry into a surfaced based one.
+//
+// The obviously smallest expansion/extrusion is condensed to the epsilon
+// thin surface.
+struct Geant4ShapeConverter {
+  /// A scale between Geant4 and ACTS
+  ActsScalar scale = 1.;
+  /// A description to keep the axis order, if it is set to false
+  /// cyclic re-ordering will happen, otherwise axis flip if needed in
+  /// order to keep the system right-handed
+  bool keepAxisOrder = false;
+
+  /// @brief Convert to cylinder bounds
+  ///
+  /// @param g4Tubs a Geant4 tube shape
+  ///
+  /// @return an Acts Cylinder bounds object
+  std::tuple<std::shared_ptr<CylinderBounds>, ActsScalar> cylinderBounds(
+      const G4Tubs& g4Tubs);
```

Comment:
what does the scalar mean in the tuple?

Reply:
Updated the documentation, the scalar is the original thickness, one needs to know that to do the material compression.